### PR TITLE
Updates for PhysX4/5 split

### PR DESCRIPTION
**Pull request by spham-amzn**
This updates the robot vacuum project to reference 'PhysX5' instead of 'PhysX' for simulation, now that both gems will be available at the same time (without the need of a cmake flag when creating the project from source). (Refer to https://github.com/o3de/o3de/pull/17436 ). Also updates need to match the development version of the ROS2 gem

### CODE REVIEW

**Review by jhanca-robotecai — CHANGES_REQUESTED** [review]
We should create a `development` branch and target all ongoing work over there before we make the `main` unusable. Unfortunately, I am not allowed to do it myself.

**Comment by spham-amzn** [comment]
> We should create a `development` branch and target all ongoing work over there before we make the `main` unusable. Unfortunately, I am not allowed to do it myself.

I agree, created a development branch (now the default) and retargeted this PR.

**Review by lemonade-dm — APPROVED** [review]

**Review by jhanca-robotecai — APPROVED** [review]
Tested with the most recent o3de and o3de-extras as of now (39bbf9b73a and 2f9a8a71fa), with d09e050 cherry-picked from #33 and it works as expected.

I would suggest merging #33 first and this one right after.

**Comment by jhanca-robotecai** [comment]
@spham-amzn 
one comment to our offline discussion: we tested the Python script to convert the prefabs and it worked as expected; the verbose information is also enough. The only thing that might cause the issues is AssetProcessor, which sometimes does not repreocess the prefabs automatically....

File: project.json [hunk 1]
```diff
@@ -18,8 +18,24 @@
...
     "restricted": "RobotVacuumSample",
     "gem_names": [
-        "ROS2",
-        "ArchVis"
+        "ArchVis",
+        "Atom",
+        "CameraFramework",
+        "DebugDraw",
+        "GameState",
+        "ImGui",
```
Comment (michalpelka) [inline project.json:26]: I believe that :
ImGui, LandscapeCanvas, SaveData, ScriptCanvasPhysics, WhiteBox are not needed.

**Review by michalpelka — DISMISSED** [review]

File: project.json (hunk 1 above)
Comment (michalpelka) [inline project.json:26]: I believe that :
ImGui, LandscapeCanvas, SaveData, ScriptCanvasPhysics, WhiteBox are not needed.

**Pull request merged**